package.json

```diff
@@ -2,6 +2,7 @@
   "version": "4.2.0",
   "private": true,
   "scripts": {
+    "postinstall": "node scripts/postinstall.js",
```

Comment:
I would rather only use this as a CI check. An install should never mutate lockfiles (see npm issues for new contributors).

Why would we need to run this on every install?

Reply:
>Why would we need to run this on every install?

The hope was to prevent cases like this https://github.com/mui-org/material-ui/pull/16521#discussion_r301615614 i'm fine either way

Reply:
Sure, I wasn't doubting the use case. Just that I don't want to cause any confusion when installing for the first time. I don't want to find out how this behaves across different versions of yarn installed. A CI task (maybe with a "how to fix" section in CONTRIBTUNG.md) should prevent those issues from being merged?

Reply:
> I don't want to find out how this behaves across different versions of yarn installed

There shouldn't be any, because the installed version doesn't matter to the results from `yarn-deduplicate`.

> Just that I don't want to cause any confusion when installing for the first time

The way it's set up right now, nothing will happen on install. It will run the script, see no duplicated packages and exit. The only time it will do anything is after `yarn add`. Since `yarn upgrade ...` doesn't trigger `postinstall` you'll only see something on the CI or if you run the script manually.
Since the CI will fail if there are duplicated packages you'll never accidentally merge with duplicated packages, reinforcing
> nothing will happen on install

Reply:
> The way it's set up right now, nothing will happen on install. It will run the script, see no duplicated packages and exit.

What if it does that is my whole point.

> here shouldn't be any

This isn't sufficient in the current state of npm/yarn + lockfiles.

Just move it into a CI task. This is the intention of the task (find problems when integrating changes) so it should live where it's intention lies.

Unless there are actual reasons to put it into postinstall. So far I've only hear "this should be fine".

Reply:
I totally agree.